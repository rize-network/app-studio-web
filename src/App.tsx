import React, { Suspense } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { ComponentRouter } from 'src/routes/routes';
import { View, Text } from 'app-studio';

export const App: React.FC = () => {
  const location = useLocation();
  return (
    <Suspense
      fallback={
        <View
          height="100vh"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Text>Loading...</Text>
        </View>
      }
    >
      <Routes location={location}>
        {ComponentRouter.map((item, index) => (
          <Route key={index} path={item.path} element={item.element} />
        ))}
      </Routes>
    </Suspense>
  );
};

export default App;
