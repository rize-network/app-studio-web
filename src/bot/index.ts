import { Bot } from './Bot';
import { DocuCode } from './DocuCode';

async function main() {
  const projectName = 'MDX Documentation';
  const descriptionPath = 'docs/description.md';

  // Get componentName from command line arguments
  const componentName = process.argv[2]; // Assuming it's the first argument passed
  if (!componentName) {
    console.error('Please provide a component name as an argument.');
    process.exit(1);
  }

  console.log(`Generating documentation for component: ${componentName}`);

  const basePath = 'src/components';
  const propsPath = `src/data/props/${componentName}.props.json`;
  const componentFolder = `${basePath}/${componentName}`;
  const componentPath = `${basePath}/${componentName}/${componentName}`;

  const assistantGPT = new Bot(componentName);

  // Implementation for Documentation of the code
  console.log('Initializing documentation process...');
  const docuCode = new DocuCode(componentName);
  const assistantDocumentation = await assistantGPT.init(
    projectName,
    [],
    descriptionPath,
    'docu'
  );

  await docuCode.processDirectory(componentFolder, assistantDocumentation.id);

  // Generating props file for code
  console.log('Generating props file...');
  const { fileIds } = await assistantGPT.addFiles(componentPath);

  const assistantCreation = await assistantGPT.init(
    projectName,
    fileIds,
    descriptionPath,
    'props'
  );

  await assistantGPT.response(assistantCreation.id, propsPath);

  await assistantGPT.MarkdownGeneration(
    componentFolder,
    componentName,
    propsPath
  );

  console.log('Documentation generation completed.');
}

main().catch((error) => {
  console.error('Error during the execution of main:', error);
});
