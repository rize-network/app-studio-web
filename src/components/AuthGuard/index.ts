export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
