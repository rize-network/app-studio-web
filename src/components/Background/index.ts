export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  WallProps,
  ParticlesProps,
  GridProps,
  RipplesProps,
  BackgroundImageProps,
  BackgroundVideoProps,
  BackgroundGradientProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
  BackgroundSize,
  BackgroundPosition,
  BackgroundRepeat,
  BlendMode,
} from './Background/Background.type';
