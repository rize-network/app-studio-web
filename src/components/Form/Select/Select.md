### **Import**

```tsx static
import { Select } from "app-studio";
```

### **Default**

```tsx
<Select label="Select an item" options={["Item1", "Item2", "Item3"]} />
```

### **Disabled**

“**_isDisabled_**” makes the select field unclickable.

```tsx
<Select isDisabled options={["Item1", "Item2", "Item3"]} />
```

### **IsReadOnly**

“**_isReadOnly_**” makes the field readable but not writable. Can only read the element.

```tsx
<Select options={["Item1", "Item2", "Item3"]} isReadOnly />
```

### **Sizes**

“**_size_**” changes the text and icon size . It has a default value of “md”.

```tsx
import { Vertical } from "../../Layout/Vertical/Vertical";

const options = ["Item1", "Item2", "Item3"];

<Vertical gap={10}>
  <Select name="xs" size="xs" placeholder="xs" options={options} />
  <Select name="sm" size="sm" placeholder="sm" options={options} />
  <Select name="md" size="md" placeholder="md" options={options} />
  <Select name="lg" size="lg" placeholder="lg" options={options} />
  <Select name="xl" size="xl" placeholder="xl" options={options} />
</Vertical>;
```

### **ColorScheme**

“**_colorScheme_**” modifies the border color of the select field.

```tsx
import { Vertical } from "../../Layout/Vertical/Vertical";

<Vertical gap={15}>
  {["theme.primary", "theme.secondary", "error", "success", "warning"].map((color) => (
    <Select
      key={color}
      options={["Item1", "Item2", "Item3"]}
      colorScheme={color}
    />
  ))}
</Vertical>;
```

### **Shadow**

“**_shadow_**” adds a shadow effect to the field.

```tsx
<Select
  shadow={{
    boxShadow:
      "rgb(204, 219, 232) 3px 3px 6px 0px inset, rgba(255, 255, 255, 0.5) -3px -3px 6px 1px inset",
  }}
  options={["Item1", "Item2", "Item3"]}
/>
```

### **Multiple**

“**_isMulti_**” indicates whether the select component allows the selection of multiple options.

```tsx
<Select
  options={["Item1", "Item2", "Item3"]}
  placeholder="Select an item.."
  isMulti
/>
```

### **Error**

“**error**” indicates whether the field has an error.

```tsx
<Select options={["Item1", "Item2", "Item3"]} error />
```

### **HelperText**

“**_helperText_**” provides additional information about the field.

```tsx
<Select
  options={["Item1", "Item2", "Item3"]}
  helperText="select one item!"
  error
/>
```

### **Styles**

“**_styles_**” changes the style of the select field.

```tsx
<Select
  colorScheme="theme.primary"
  options={["Item1", "Item2", "Item3"]}
  styles={{
    selectBox: { borderRadius: 10,  borderColor: 'theme.primary',
        borderStyle: 'solid',
        borderWidth: 1, },
    text: { color: "theme.primary"  },
    label: { fontWeight: "bold", color: "theme.primary"  },
  }}
/>
```

### **Default Value**

```tsx static
const Sizes: Record<Size, CSSProperties> = {
  xs: { height: 6, width: 6 },
  sm: { height: 12, width: 12 },
  md: { height: 18, width: 18 },
  lg: { height: 24, width: 24 },
  xl: { height: 30, width: 30 },
};
```

```tsx static
const IconSizes: Record<Size, string> = {
  xs: "8px",
  sm: "10px",
  md: "12px",
  lg: "14px",
  xl: "16px",
};
```

### Types

```tsx static
type Size = "xs" | "sm" | "md" | "lg" | "xl";
```

```tsx static
type SelectStyles = {
  box?: CSSProperties;
  text?: CSSProperties;
  label?: CSSProperties;
  helperText?: CSSProperties;
  field?: CSSProperties;
};
```
