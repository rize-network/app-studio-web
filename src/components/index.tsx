export * from './Button/Button';
export * from './Form/Checkbox/Checkbox';
export * from './Form/CountryPicker/CountryPicker';
export * from './Form/DatePicker/DatePicker';
export * from './Form/Password/Password';
export * from './Form/Select/Select';
export * from './Form/Switch/Switch';
export * from './Form/Switch/Switch';
export * from './Form/TextArea/TextArea';
export * from './Form/TextField/TextField';
export * from './Layout/Center/Center';
export * from './Formik';
export * from './Layout/Horizontal/Horizontal';
export * from './Layout/Vertical/Vertical';
export * from './Layout/View/View';
export * from './Link/Link';
export * from './Loader/Loader';
export * from './Modal/Modal';
export * from './Text/Text';
export * from './File/File';
export * from './Tabs/Tabs';
export * from './Alert/Alert';
export * from './AspectRatio/AspectRatio';
export * from './Avatar/Avatar';
export * from './Badge/Badge';
export * from './Toggle/Toggle';
export * from './ToggleGroup/ToggleGroup';
export * from './Message/Message';
