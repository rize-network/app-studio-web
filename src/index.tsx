import React from 'react';
import ReactDOM from 'react-dom/client';
import AppProvider from 'src/providers';
import { App } from './App';
import { ComboBoxProvider } from './components/Form/ComboBox/ComboBox/ComboBox.provider';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <AppProvider>
      <ComboBoxProvider>
        <App />
      </ComboBoxProvider>
    </AppProvider>
  </React.StrictMode>
);
