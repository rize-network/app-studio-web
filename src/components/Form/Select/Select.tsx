import React from 'react';

import { SelectProps } from './Select/Select.props';
import { useSelectState } from './Select/Select.state';
import SelectView from './Select/Select.view';

const SelectComponent: React.FC<SelectProps> = (props) => {
  const selectStates = useSelectState(props);
  return <SelectView {...selectStates} {...props} />;
};

/**
 * Select provides a dropdown list of options for the user to choose from.
 */
export const Select = SelectComponent;
