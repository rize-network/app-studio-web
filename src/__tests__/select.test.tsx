import React from 'react';
import renderer from 'react-test-renderer';
import { Select } from 'src/components';
import { MultiSelect } from 'src/components/Form/Select/Select/Select.view';

import { cleanup, fireEvent, render, screen } from '@testing-library/react';

afterEach(() => {
  cleanup();
});

describe('Select component', () => {
  test('renders Select component', () => {
    render(<Select name="active" role="textbox" options={['Item1', 'Item2', 'Item3']} />);
    const SelectElement = screen.getByRole('textbox');
    expect(SelectElement).toBeInTheDocument();
  });

  test('renders disabled Select', () => {
    render(<Select name="diabled" isDisabled role="textbox" options={['Item1', 'Item2', 'Item3']} />);
    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toBeDisabled();
  });

  test('renders readonly Select', () => {
    render(<Select name="username" isReadOnly role="textbox" options={['Item1', 'Item2', 'Item3']} />);
    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toHaveAttribute('readonly');
  });

  test('cannot change state when readOnly prop is passed', () => {
    render(
      <Select isReadOnly role="textbox" options={['Item1', 'Item2', 'Item3']}>
        Test Select
      </Select>
    );
    const select = screen.getByRole('textbox');
    fireEvent.click(select);
    expect(select).not.toBeChecked();
  });

  test('Select to match snapshot', () => {
    const tree = renderer
      .create(
        <Select
          name="name"
          size="xl"
          error
          options={['Item1', 'Item2', 'Item3']}
          shadow={{ boxShadow: 'rgb(249, 115, 22) 0px 4px 14px 0px' }}
          isReadOnly
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MultiSelect component', () => {
  const option = 'Option 1';

  test('renders MultiSelect component without crashing', () => {
    render(<MultiSelect option={option} removeOption={() => {}} />);
    expect(screen.getByText(option)).toBeInTheDocument();
  });

  test('calls removeOption when close button is clicked', () => {
    const option = 'Option 1';
    const removeOption = jest.fn();
    render(<MultiSelect option={option} removeOption={removeOption} />);

    const closeButton = screen.getByRole('close-button');
    fireEvent.click(closeButton);
    expect(removeOption).toHaveBeenCalledWith(option);
  });

  test('renders MultiSelect component with provided size', () => {
    render(<MultiSelect option={option} role="MultiSelect" size="md" removeOption={() => {}} />);
    const MultiSelectElement = screen.getByRole('MultiSelect');
    expect(MultiSelectElement).toBeInTheDocument();
    expect(MultiSelectElement).toHaveStyle({ fontSize: 16 });
  });
});
