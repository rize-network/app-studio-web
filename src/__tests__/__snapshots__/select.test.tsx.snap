// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component Select to match snapshot 1`] = `
<div
  className="sc-bdfCDU krrJWc"
  display="flex"
  flexDirection="column"
  flexWrap="nowrap"
  gap={5}
  justifyContent="flex-start"
  onClick={[Function]}
  position="relative"
  role="SelectBox"
>
  <div
    alignItems="center"
    backgroundColor="color.white"
    borderBottomWidth={1}
    borderColor="error"
    borderRadius={0}
    borderStyle="solid"
    borderWidth={0}
    boxShadow="rgb(249, 115, 22) 0px 4px 14px 0px"
    className="sc-bdfCDU krrJWc"
    color="theme.primary"
    cursor="auto"
    display="flex"
    flexDirection="row"
    flexWrap="nowrap"
    fontSize={20}
    gap={10}
    justifyContent="space-between"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    paddingBottom={4}
    paddingLeft={16}
    paddingRight={36}
    paddingTop={10}
    transition="padding 0.2s ease"
    width="100%"
  >
    <div
      className="sc-bdfCDU krrJWc"
      display="flex"
      flexDirection="column"
      flexWrap="nowrap"
      justifyContent="flex-start"
      onClick={[Function]}
      width="100%"
    >
      <select
        className="sc-bdfCDU krrJWc"
        disabled={false}
        height={0}
        multiple={false}
        name="name"
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        opacity={0}
        position="absolute"
        readOnly={true}
        value="Item1"
        width={0}
      >
        <option
          value="Item1"
        >
          Item1
        </option>
        <option
          value="Item2"
        >
          Item2
        </option>
        <option
          value="Item3"
        >
          Item3
        </option>
      </select>
      <div
        backgroundColor="transparent"
        border="none"
        className="sc-bdfCDU iBzhpA"
        color="color.blueGray.700"
        cursor="auto"
        fontSize={20}
        fontStyle="normal"
        fontWeight={400}
        heigth="100%"
        lineHeight={14}
        margin={0}
        onClick={[Function]}
        paddingHorizontal={0}
        paddingVertical={8}
        role="text"
        textDecoration="none"
        width="95%"
      >
        Item1
      </div>
    </div>
    <div
      alignItems="center"
      className="sc-bdfCDU krrJWc"
      display="flex"
      flexWrap="nowrap"
      gap={10}
      justifyContent="center"
      onClick={[Function]}
      position="absolute"
      right={16}
      zIndex={500}
    />
  </div>
  <div
    className="sc-bdfCDU krrJWc"
    color="theme.error"
    fontSize={10}
    fontStyle="normal"
    fontWeight={400}
    lineHeight={10}
    marginHorizontal={0}
    marginVertical={0}
    onClick={[Function]}
    role="text"
    textDecoration="none"
  />
</div>
`;
